admin/src/main/java/edu/unc/lib/dl/admin/controller/IngestSourceController.java

```diff
@@ -0,0 +1,188 @@
+		result.put("candidates", sourceManager.listCandidates(destination));
+		
+		return result;
+	}
+
+	@RequestMapping(value = "ingestFromSource/{pid}", method = RequestMethod.POST, produces = "application/json")
+	public @ResponseBody Object ingestFromSource(@PathVariable("pid") String pid,
+			@RequestBody List<IngestPackageDetails> packages, HttpServletResponse resp) {
+		
+		log.error("Request to ingest from source to {}", pid);
+		PID destPid = new PID(pid);
+
+		// Check that user has permission to deposit to the selected destination
+		if (!aclService.hasAccess(destPid, GroupsThreadStore.getGroups(), Permission.addRemoveContents)) {
+			log.debug("Access denied to user {} while attempting to from ingest source deposit to {}",
+					GroupsThreadStore.getUsername(), pid);
+			resp.setStatus(HttpServletResponse.SC_FORBIDDEN);
+			return "Insufficient permissions to deposit to the selected destination";
+		}
+		
+		// Validate the packages requested for deposit
+		for (IngestPackageDetails packageDetails : packages) {
+			if (StringUtils.isBlank(packageDetails.getPackagePath())
+					|| StringUtils.isBlank(packageDetails.getPackagingType())) {
+				resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+				return "Package selected for deposit was missing either a path or packaging type";
+			}
+			
+			// Verify that the package path is from within the allowed locations for the specified ingest source
+			if (!sourceManager.isPathValid(packageDetails.getPackagePath(), packageDetails.getSourceId())) {
+				resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+				return "Invalid source path specified: " + packageDetails.getPackagePath();
+			}
+		}
+		
+		// Build deposit entries and add to queue
+		for (IngestPackageDetails packageDetails : packages) {
+			IngestSourceConfiguration source = sourceManager.getSourceConfiguration(packageDetails.getSourceId());
+			
+			PID depositPID = new PID("uuid:" + UUID.randomUUID().toString());
+			
+			// Generate a filename if one was not provided
+			String filename = packageDetails.getLabel();
+			if (StringUtils.isBlank(filename)) {
+				filename = packageDetails.getPackagePath().substring(packageDetails.getPackagePath().lastIndexOf('/') + 1);
+			}
+			
+			Map<String, String> deposit = new HashMap<>();
+			deposit.put(DepositField.sourcePath.name(), source.getBase() + packageDetails.getPackagePath());
+			deposit.put(DepositField.fileName.name(), packageDetails.getLabel());
+			deposit.put(DepositField.packagingType.name(), packageDetails.getPackagingType());
+			deposit.put(DepositField.uuid.name(), depositPID.getUUID());
+			deposit.put(DepositField.submitTime.name(), String.valueOf(System.currentTimeMillis()));
+			deposit.put(DepositField.depositorName.name(), GroupsThreadStore.getUsername());
+			deposit.put(DepositField.permissionGroups.name(), GroupsThreadStore.getGroupString());
+			deposit.put(DepositField.depositorEmail.name(), GroupsThreadStore.getUsername() + "@email.unc.edu");
```

Comment:
We get email addresses in two ways across the CDR codebase: by appending "@email.unc.edu" to the onyen and by looking at the "mail" request header. Can we settle on one of these?